File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/SiteInfoUseCase.java
package com.glia.widgets.chat.domain;

import androidx.annotation.NonNull;

import com.glia.androidsdk.RequestCallback;
import com.glia.androidsdk.site.SiteInfo;
import com.glia.widgets.core.engagement.GliaEngagementRepository;

public class SiteInfoUseCase {
    private final GliaEngagementRepository repository;

    public SiteInfoUseCase(GliaEngagementRepository repository) {
        this.repository = repository;
    }

    public void execute(@NonNull RequestCallback<SiteInfo> callback) {
        repository.getSiteInfo(callback);
    }

}

File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/SiteInfoUseCase.kt
package com.glia.widgets.chat.domain

import com.glia.androidsdk.RequestCallback
import com.glia.androidsdk.site.SiteInfo
import com.glia.widgets.di.GliaCore

internal class SiteInfoUseCase(private val gliaCore: GliaCore) {
    operator fun invoke(callback: RequestCallback<SiteInfo?>) = gliaCore.getSiteInfo(callback)
}
